internal/elasticsearch/transform/transform.go

```diff
@@ -0,0 +1,548 @@
+			if err := json.NewDecoder(strings.NewReader(v.(string))).Decode(&runtimeMappings); err != nil {
+				return nil, err
+			}
+			transform.Source.RuntimeMappings = runtimeMappings
+		}
+	}
+
+	if v, ok := d.GetOk("destination"); ok {
+
+		definedDestination := v.([]interface{})[0].(map[string]interface{})
+		transform.Destination = new(models.TransformDestination)
+
+		transform.Destination.Index = definedDestination["index"].(string)
+
+		if pipeline, ok := definedDestination["pipeline"]; ok && len(pipeline.(string)) > 0 {
+			transform.Destination.Pipeline = pipeline.(string)
+		}
+	}
+
+	if v, ok := d.GetOk("pivot"); ok {
+		var pivot interface{}
+		if err := json.NewDecoder(strings.NewReader(v.(string))).Decode(&pivot); err != nil {
+			return nil, err
+		}
+		transform.Pivot = pivot
+	}
+
+	if v, ok := d.GetOk("latest"); ok {
+		var latest interface{}
+		if err := json.NewDecoder(strings.NewReader(v.(string))).Decode(&latest); err != nil {
+			return nil, err
+		}
+		transform.Latest = latest
+	}
+
+	if v, ok := d.GetOk("frequency"); ok {
+		transform.Frequency = v.(string)
+	}
+
+	if v, ok := d.GetOk("metadata"); ok {
+		metadata := make(map[string]interface{})
+		if err := json.NewDecoder(strings.NewReader(v.(string))).Decode(&metadata); err != nil {
+			return nil, err
+		}
+		transform.Meta = metadata
+	}
+
+	if v, ok := d.GetOk("retention_policy"); ok && v != nil {
+		definedRetentionPolicy := v.([]interface{})[0].(map[string]interface{})
+		retentionTime := models.TransformRetentionPolicyTime{}
+		if v, ok := definedRetentionPolicy["time"]; ok {
+			var definedRetentionTime = v.([]interface{})[0].(map[string]interface{})
+			if f, ok := definedRetentionTime["field"]; ok {
+				retentionTime.Field = f.(string)
+			}
+			if ma, ok := definedRetentionTime["max_age"]; ok {
+				retentionTime.MaxAge = ma.(string)
+			}
+			transform.RetentionPolicy = new(models.TransformRetentionPolicy)
+			transform.RetentionPolicy.Time = retentionTime
+		}
+	}
+
+	if v, ok := d.GetOk("sync"); ok {
+		definedRetentionPolicy := v.([]interface{})[0].(map[string]interface{})
+		syncTime := models.TransformSyncTime{}
+		if v, ok := definedRetentionPolicy["time"]; ok {
+			var definedRetentionTime = v.([]interface{})[0].(map[string]interface{})
+			if f, ok := definedRetentionTime["field"]; ok {
+				syncTime.Field = f.(string)
+			}
+			if d, ok := definedRetentionTime["delay"]; ok {
+				syncTime.Delay = d.(string)
+			}
+			transform.Sync = new(models.TransformSync)
+			transform.Sync.Time = syncTime
+		}
+	}
+
+	if v, ok := d.GetOk("settings"); ok {
+		definedSettings := v.([]interface{})[0].(map[string]interface{})
+		settings := models.TransformSettings{}
+		if v, ok := definedSettings["align_checkpoints"]; ok {
+			settings.AlignCheckpoints = new(bool)
+			*settings.AlignCheckpoints = v.(bool)
+		}
+		if v, ok := definedSettings["dates_as_epoch_millis"]; ok {
+			settings.DatesAsEpochMillis = new(bool)
+			*settings.DatesAsEpochMillis = v.(bool)
```

Comment:
```suggestion
			d := v.(bool)
			*settings.DatesAsEpochMillis = &d
```

and similarly for below